File: src/main/java/ru/javarush/cryptoanalyzer/konovalov/view/Menu.java
package ru.javarush.cryptoanalyzer.konovalov.view;

import java.util.Scanner;

import static ru.javarush.cryptoanalyzer.konovalov.view.Messages.*;

public class Menu {
    private final Scanner SCANNER;

    public Menu(Scanner scanner) {
        this.SCANNER = scanner;
    }

    public String[] getArgs() {
        int mode = getMode(SCANNER);
        String [] args = new String[QUESTIONS[mode].length];
        args[0] = QUESTIONS[mode][0][0];
        for (int i = 1; i < args.length; i++) {
            String quest = QUESTIONS[mode][i][0];
            System.out.println(quest);
            String answer = SCANNER.nextLine();
            args[i] = "".equals(answer.trim()) ? QUESTIONS[mode][i][1] : answer;
        }
            return args;
    }

    private int getMode(Scanner scanner) {
        int mode;

        do {
            System.out.println(MESSAGE_SELECT_MODE);
            String input = scanner.nextLine();
            mode = switch (input) {
                case "1" -> 0;
                case "2" -> 1;
                case "3" -> 2;
                case "4" -> 3;
                case "5" -> 4;
                default -> {
                    System.out.println(INCORRECT_SELECTION_MODE);
                    yield -1;
                }
            };
        } while (mode < 0);
        return mode;


    }
}

File: src/main/kotlin/view/Menu.kt
package view

import java.util.*

class Menu(private val scanner: Scanner) {

    fun getArgs(): Array<String> {
        val modeId : Int = getMode(scanner)
        val mode : Array<Array<String>> = QUESTIONS[modeId]
        val nullArgs: Array<String?> = arrayOfNulls(mode.size)
        val args: Array<String> =nullArgs.map { it-> ""}.toTypedArray()

        val actionType = mode[0][0]
        args[0] = actionType

        for (i in 1 until args.size) {
            val question = mode[i][0]
            val defaultAnswer = mode[i][1]
            println(question)
            val userAnswer: String = scanner.nextLine()
            val isEmptyAnswer : Boolean = (userAnswer == "")
            args[i] = if (isEmptyAnswer) defaultAnswer else userAnswer
        }

        return args
    }

    private fun getMode(scanner: Scanner) : Int{
        var mode: Int

        do{
            println(MESSAGE_SELECT_MODE)
            val input: String? = scanner.nextLine()
            mode = when(input){
                "1" -> 0;
                "2" -> 1;
                "3" -> 2;
                "4" -> 3;
                "5" -> 4;
                else -> {
                    println(INCORRECT_SELECTION_MODE)
                    -1
                }
            }
        }
            while(mode<0)
        return mode
    }

}